### PR TITLE
DEV: Fetcher and benchmark fixes

### PR DESCRIPTION
**Pull request by jbredeche**
Also, moved a lot of Fetcher test into Zipline.

### CODE REVIEW

File: zipline/utils/test_utils.py [hunk 1]
```diff
@@ -741,6 +741,30 @@ def __init__(self):
...
+class FetcherDataPortal(DataPortal):
+    def __init__(self, env, sim_params):
+        super(FetcherDataPortal, self).__init__(
+            env,
+            sim_params
+        )
+
+    def get_spot_value(self, asset, field, dt=None):
+        # if this is a fetcher field, exercise the regular code path
+        if self._check_fetcher(asset, field, (dt or self.current_dt)):
```
Comment (ssanderson) [inline zipline/utils/test_utils.py]: Where is _check_fetcher defined?

Comment (jbredeche) [inline zipline/utils/test_utils.py]: in `DataPortal`

Comment (ssanderson) [inline zipline/utils/test_utils.py]: For some reason I thought this PR was against master.  Apologies.  Carry on.

File: zipline/utils/test_utils.py [hunk 2]
```diff
@@ -741,6 +741,30 @@ def __init__(self):
...
+            return super(FetcherDataPortal, self).get_spot_value(
+                asset, field, dt)
+
+        # otherwise just return a fixed value
+        return int(asset)
+
+    def _get_daily_window_for_sid(self, asset, field, days_in_window,
+                                  extra_slot=True):
+        return np.array(range(0, len(days_in_window)), dtype=np.float64)
```
Comment (ssanderson) [inline zipline/utils/test_utils.py]: This probably wants to be `arange(days_in_window, dtype=np.float64)`.

Comment (jbredeche) [inline zipline/utils/test_utils.py]: Agreed, thanks.

**Pull request merged**